src/figuresClasses.ts

```diff
@@ -1,11 +1,74 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'blue' | 'green' | 'red';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a + b <= c || a + c <= b || b + c <= a) {
+      throw new Error('Invalid triangle dimensions');
+    }
+
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Invalid side lengths');
+    }
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+
+    return parseFloat(
+      Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c)).toFixed(2),
+    );
+  }
+}
 
-export class Circle implements Figure {}
+export class Circle implements Figure {
+  shape: Shape = 'circle';
 
-export class Rectangle implements Figure {}
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
```

Comment:
[CODE STYLE] - The error message 'Invalid radius' should be more descriptive. For example, 'Circle radius must be greater than zero.'